### PR TITLE
Autocomplete: Fix Showing After Unfocus

### PR DESCRIPTION
**Pull request by crhallberg**
Fixes a bug discovered by @EreMaijala. Thank you!

Removed this debounce-caused race condition:

1. start a search.
1. input event adds a 300ms delay to debounce the search.
1. tab away, clearing all callbacks.
1. a brand new search begins as the delay ends, setting `lastCB`.
1. callback matches and shows results.

### CODE REVIEW

**Comment by crhallberg** [comment]
Question: once this passes, do we want to remove the vendor file since we have a node_modules copy task for this?

**Comment by demiankatz** [comment]
@crhallberg, no, we want to continue to commit the vendor JS files -- we want people to be able to check out the repo and run things without having to set up Node. The node_modules copy task is for use by the dev team to update the committed files; it's not intended for use by end users. (This is similar to how we commit compiled.css so people don't have to run LESS compilers on their own).

File: themes/bootstrap3/js/vendor/autocomplete.js
Comment (demiankatz) [inline themes/bootstrap3/js/vendor/autocomplete.js]: We'll need to update this when we mint the release. I'm sure you already know that, but just putting this reminder here anyway. :-)

**Comment by demiankatz** [comment]
This is working well in my test environment, and all tests are passing there. I've just triggered a Jenkins build against this branch to see if things pass there as well. I'll approve if they do!

**Comment by demiankatz** [comment]
Unfortunately, this is still failing in Jenkins:

```
1) VuFindTest\Mink\AutocompleteTest::testMultipleAutocompletesInSingleSession
Failed after 6 tries, with 9 seconds snooze time.
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'JSTOR (Organization)'
+'Loading…'
```

**Comment by demiankatz** [comment]
I don't think it tells us very much, but here's the screenshot from the broken Jenkins test. Autocomplete is not visible at all, but that's probably not surprising, since the screen shot is not taken until after the test fails.

testMultipleAutocompletesInSingleSession-2962330908622200

**Comment by demiankatz** [comment]
I've just opened #3414 to make some improvements to the test suite, in case the problem is somehow test-related and not subject-related. Once that is merged, I'll re-run the experiment. But in the meantime, I think it's also worth studying the Javascript some more, since I'm not especially optimistic that the test changes will fix this.

File: themes/bootstrap3/js/vendor/autocomplete.js
Comment (EreMaijala) [inline themes/bootstrap3/js/vendor/autocomplete.js]: I don't think we can return without calling the callback here if there are no items. Otherwise the "Loading..." text will remain on the screen indefinitely.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: themes/bootstrap3/js/vendor/autocomplete.js
Comment (crhallberg) [reply themes/bootstrap3/js/vendor/autocomplete.js]: Is there a call to `_hide()` missing here?

Comment (demiankatz) [reply themes/bootstrap3/js/vendor/autocomplete.js]: I don't _think_ a call to `_hide()` should hurt anything here, and it seems like it's probably the correct path forward. @EreMaijala, before we make further changes, do you have a specific reproducible problem related to this, or is it just a general observation? Seems like it ought to be possible to cause the "Loading" indicator to get stuck if I'm understanding correctly, though I've not seen that problem occur in my own testing so far. (I didn't have time to try again today, though).

Comment (EreMaijala) [reply themes/bootstrap3/js/vendor/autocomplete.js]: Just enter something that doesn't yield any suggestions. Then the items length is 0 and "Loading..." stays on screen.
I also think that it would be good to remove the "Loading..." content in _hide(). This would at least ensure we don't get confused if a test fails.

Comment (demiankatz) [reply themes/bootstrap3/js/vendor/autocomplete.js]: @crhallberg and @EreMaijala, please see commit b84fa55e0e640fc16c7317e039b55e12122d7a52 -- I have implemented @EreMaijala's suggestions, plus I have moved the logic to display the "Loading..." message. I noticed that as written, we were showing a loading message even for inputs that were too short to trigger actual loading. Now we only say "Loading" while we're actually loading something. This eliminates unnecessary loading indicators as well as flashes of "loading" before retrieving locally-cached data. Tests are still passing. Any concerns about any of these changes?

Comment (EreMaijala) [reply themes/bootstrap3/js/vendor/autocomplete.js]: Hooray, I can't break anything and all the changes now make sense! :)

**Review by EreMaijala — APPROVED** [review]

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @EreMaijala! I'm just putting this in "request changes" status until @crhallberg releases the revised code in the autocomplete.js repo and updates the package.json here. Once all that housekeeping is done, I will approve and merge. Thanks, everyone!

**Comment by crhallberg** [comment]
Minted: https://github.com/vufind-org/autocomplete.js/releases/tag/v2.1.9

**Review by demiankatz — APPROVED** [review]
Thanks, @crhallberg -- I've updated package.json, refreshed dependencies, and confirmed that tests still pass. Merging now.

**Pull request merged**
